test/Dapr.E2E.Test.App/Actors/ReentrantActor.cs

```diff
@@ -0,0 +1,57 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Dapr.Actors.Runtime;
+
+namespace Dapr.E2E.Test.Actors.Reentrancy
+{
+    public class ReentrantActor : Actor, IReentrantActor
+    {
+        public ReentrantActor(ActorHost host)
+            : base(host)
+        {
+        }
+
+        public Task Ping()
+        {
+            return Task.CompletedTask;
+        }
+
+        public async Task ReentrantCall(ReentrantCallOptions callOptions)
+        {
+            try
+            {
+                await UpdateState(true);
+                var actor = this.ProxyFactory.CreateActorProxy<IReentrantActor>(this.Id, "ReentrantActor");
+                if (callOptions == null || callOptions.CallsRemaining <= 1)
+                {   
+                    await actor.Ping();
+                }
+                else
+                {
+                    await actor.ReentrantCall(new ReentrantCallOptions { CallsRemaining = callOptions.CallsRemaining - 1 });
+                }                
+            } 
+            finally 
+            {
+                await UpdateState(false);
+            }
+        }
+
+        public Task<State> GetState() 
+        {
+            return this.StateManager.GetOrAddStateAsync<State>("reentrant-record", new State());
+        }
+
+        private async Task UpdateState(bool isEnter)
```

Comment:
What is `isEnter` mean/do?

Reply:
Notice that it's hardcoded to `true` below

Reply:
This is following the pattern for other e2e tests involving actors. Though in this case I forgot to assign it :|

Essentially what it lets us do is just track when an actor's method is entered and left. This let's us see the single threaded path.